src/test/README.md

```diff
@@ -5,7 +5,10 @@ and tests weren't explicitly disabled.
 
 After configuring, they can be run with `make check`.
 
-To run the bitcoind tests manually, launch `src/test/test_bitcoin`.
+To run the bitcoind tests manually, launch `src/test/test_bitcoin`. To recompile
+after a test file was modified, run `make` and then run the test again. If you
+modify a non-test file, use `make -C src/test` to recompile only what's needed
+to run the bitcoind tests.
```

Comment:
I'm a bit confused by this description...excuse me for asking some beginner questions here...😂 
I couldn't find a lot of info on how to use make and `-C`, but I did find one here. According to this [post](https://www.gnu.org/software/make/manual/make.html#Phony-Targets) about phony targets, I derived that the dir after -C flag is the target subdir to build:

> Phony targets are also useful in conjunction with recursive invocations of make (see Recursive Use of make). In this situation the makefile will often contain a variable which lists a number of sub-directories to be built. A simplistic way to handle this is to define one rule with a recipe that loops over the sub-directories, like this:
> 
> SUBDIRS = foo bar baz
> 
> subdirs:
>         for dir in $(SUBDIRS); do \
>           $(MAKE) -C $$dir; \
>         done
> 

If so, `make -C src/test` is only building the test directory, right? Why is it that if one `modify a non-test file`, one needs to rebuild the `src/test`? They should not have to rebuild `src/test` if they didn't modify any test file?

Reply:
the test depend on bitcoin core, so if a non-test file (a bitcoin core file) like init.cpp is modified, it will need to be recompiled as well.

init.cpp is then put into libbitcoin_server, which is then linked into test_bitcoin

Reply:
Sounds like either the a test or a non-test file gets changed, bitcoin core (the src dir?) needs to be recompiled. Then when is `make -C src/test` relevant? Because this command only compiles what is in `src/test` right? The documentation says `If you
modify a non-test file, use `make -C src/test` to recompile only what's needed
to run the bitcoind tests`, and it seems to imply that when a non-test file gets changed, we only need to compile `src/test`?

Reply:
Tests depend on individual source files, not all of `bitcoind`. So if you're working on `src/descriptor.cpp` it's faster to do `make -C src/test && src/test/test_bitcoin  --run_test=descriptor_tests`

Reply:
> Tests depend on individual source files, not all of bitcoind

I don't think this is true. Tests depend on all the libraries that bitcoind depends on. The only thing they don't depend on is bitcoind.cpp